backend/taxHandlers.ts

```diff
@@ -0,0 +1,259 @@
+import { DEFAULT_TAX_CODE } from "@/constants";
+import {
+  OrderSubscriptionFragment,
+  TaxBaseSubscriptionFragment,
+} from "../generated/graphql";
+import { fetchTaxes } from "./avataxApi";
+
+import {
+  FetchTaxesLinePayload,
+  ResponseTaxPayload,
+  AvataxConfig,
+  FetchTaxesPayload,
+} from "./types";
+import {
+  getAvataxConfig,
+  avataxConfigIsValidToUse,
+  taxObjectIsValidToUse,
+} from "./utils";
+
+const prepareLinesPayload = (
+  taxData: TaxBaseSubscriptionFragment
+): Array<FetchTaxesLinePayload> => {
+  const lines = taxData.lines;
+  const linesPayload = lines.map((line, index) => {
+    const taxCode =
+      line.sourceLine.__typename === "OrderLine"
+        ? line.sourceLine.variant!.product.metafield ||
+          line.sourceLine.variant!.product.productType.metafield
+        : line.sourceLine.productVariant.product.metafield ||
+          line.sourceLine.productVariant.product.productType.metafield;
+    const itemCode =
+      line.sourceLine.__typename === "OrderLine"
+        ? line.sourceLine.productSku || line.sourceLine.variant!.id
+        : line.sourceLine.productVariant.sku ||
+          line.sourceLine.productVariant.id;
```

Comment:
This inline condition is really hard to read - maybe refactoring this to a separate function would be better?